src/Http/GraphRequest.php

```diff
@@ -162,14 +153,18 @@ public function setReturnType(string $returnClass): self
     }
 
     /**
-    * Adds custom headers to the request
-    *
-    * @param array $headers An array of custom headers
-    *
-    * @return GraphRequest object
-    */
+     * Adds custom headers to the request
+     *
+     * @param array<string, string|string[]> $headers An array of custom headers
+     *
+     * @return GraphRequest object
+     * @throws GraphClientException if attempting to overwrite SdkVersion header
+     */
     public function addHeaders(array $headers): self
     {
+        if (array_key_exists("SdkVersion", $headers)) {
+            throw new GraphClientException("Cannot overwrite SdkVersion header");
+        }
         $this->headers = array_merge_recursive($this->headers, $headers);
```

Comment:
Any reasons for a recursive merge?

Reply:
It allows us to append multiple values to a given header for headers that support multiple values e.g. Accept, Accept-Language, Cache-Control etc

Reply:
Adding a comment for future clarity

Reply:
There's a [test](https://github.com/microsoftgraph/msgraph-sdk-php-core/blob/philip/fix/tests/tests/Http/Request/GraphRequestTest.php#L190) to cover this as well